gravitee-am-model/src/main/java/io/gravitee/am/model/FactorSettings.java

```diff
@@ -15,18 +15,28 @@
  */
 package io.gravitee.am.model;
 
-import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
+import java.util.List;
+import java.util.Objects;
+
 @Getter
 @Setter
 @NoArgsConstructor
+@AllArgsConstructor
+@Builder
 public class FactorSettings {
     private String defaultFactorId;
     private List<ApplicationFactorSettings> applicationFactors = List.of();
 
+    public List<ApplicationFactorSettings> getApplicationFactors() {
+        return Objects.requireNonNullElseGet(applicationFactors, List::of);
```

Comment:
Is it required? It can throw NullPointerException. If yes, why not: 
return applicationFactors != null ? applicationFactors : List.of();

Reply:
`requireNonNullElseGet()` does exactly what your comment says :)
The reason this safeguard is required is that when `FactorSettings` gets instantiated by deserialization, e.g. from the database, then the field defaults are ignored. 

The linked issue describes exactly such case: the application is represented in the DB as 
```
{
  "factor": {},
  ...
}
```
Deserialization effectively does the equivalent of: 
```java
var factor = new FactorSettings();
factor.setDefaultFactorId(null);
factor.setApplicationFactors(null);
```
thus leading to NPE and a 500 response.